### PR TITLE
S3 Action: Fix retrieving ipa info containing embedded app bundle

### PR DESCRIPTION
**Pull request by ngs**
Fixed retrieving `.ipa` info that contains embedded app bundle like WatchOS application.

### CODE REVIEW

File: spec/actions_specs/s3_spec.rb [hunk 1]
```diff
@@ -150,4 +150,20 @@
       end
     end
   end
+
+  if RUBY_PLATFORM =~ /darwin/i
```
Comment (KrauseFx) [inline spec/actions_specs/s3_spec.rb]: Could you replace that with `FastlaneCore::Helper.mac?` (if that works)

File: lib/fastlane/actions/s3.rb [hunk 2]
```diff
@@ -261,7 +261,9 @@ def self.get_ipa_info(ipa_file)
         bundle_id, bundle_version, title, full_version = nil
         Dir.mktmpdir do |dir|
           system "unzip -q #{ipa_file} -d #{dir} 2> /dev/null"
-          plist = Dir["#{dir}/**/*.app/Info.plist"].last
+          plist = Dir["#{dir}/**/*.app/Info.plist"].detect do|f|
+            !f.match %r{Payload/(?:.+)\.app/.+\.app/Info.plist$}
```
Comment (KrauseFx) [inline lib/fastlane/actions/s3.rb]: Would you mind moving this into `fastlane_core`? We already have some code here to retrieve this kind of data, and your solution looks pretty awesome https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/ipa_file_analyser.rb#L22

**Comment by KrauseFx** [comment]
Thanks for the pull request, I added a comment :+1:

**Comment by ngs** [comment]
@KrauseFx Thanks for feedbacks, I've sent PR on `fastlane_core`: https://github.com/fastlane/fastlane_core/pull/66

I'll rewrite this PR using `FastlaneCore::IpaFileAnalyser.fetch_info_plist_file[...]` once new `fastlane_core` version is released.

**Comment by KrauseFx** [comment]
Great, thanks @ngs. Do you want to update the `s3` action to use the class from fastlane_core?

**Comment by ngs** [comment]
@KrauseFx Updated the S3 action using `FastlaneCore::IpaFileAnalyser.fetch_info_plist_file`.

I'll revert `Gemfile` and update `fastlane.gemspec` after new version of `fastlane_core` becomes available.

**Comment by KrauseFx** [comment]
@ngs New `fastlane_core` is available :+1:

**Comment by ngs** [comment]
@KrauseFx Updated, thanks!

**Comment by KrauseFx** [comment]
Cool, thanks for squasing the commit. Did you test everything locally?

**Comment by ngs** [comment]
Yes, I'm referencing this branch from my working project: https://github.com/ngs/ci2go/blob/master/Gemfile#L11

https://circleci.com/gh/ngs/ci2go/tree/master

**Comment by KrauseFx** [comment]
Perfect, thank you so much @ngs :rocket:

**Pull request merged**